roomserver/internal/helpers/auth.go

```diff
@@ -229,6 +226,16 @@ func loadAuthEvents(
 	if result.events, err = db.Events(ctx, eventNIDs); err != nil {
 		return
 	}
+	roomID := ""
+	for _, ev := range result.events {
+		if ev.RoomID() != "" {
```

Comment:
Presumably this is meant to be `if roomID == "" { ... }` so that you only set `roomID` once?